chapters/classes.tex

```diff
@@ -340,11 +353,27 @@ \subsection{Component Variability Prefixes discrete, parameter, constant}\label{
 of initial value problem of the hybrid DAE):
 \begin{itemize}
 \item
-  A variable \lstinline!vc! declared with \lstinline!constant! prefix remains constant during transient analysis,
-  with a value that is unaffected by the initialization problem.
+  A variable \lstinline!vc! declared with \lstinline!constant! prefix remains constant during transient analysis, with a value that is unaffected by the initialization problem (i.e., determined during translation).
 \item
-  A variable \lstinline!vc! declared with the \lstinline!parameter! prefix remains constant during transient analysis,
-  with a value determined by the initialization problem.
+  % TODO: Add "structural parameter" and "evaluated parameter" to index.
+  A variable \lstinline!vs! is called a \firstuse{structural parameter} if all of the following applies:
+  \begin{itemize}
+  \item
+    It is declared with the \lstinline!parameter! prefix.
+  \item
+    It has \lstinline!fixed = true!.
+  \item
+    It does not have annotation \lstinline!Evaluate = false!.
+  \item
+    The declaration equation -- if any -- is given by a structural expression (\cref{structural-expressions}).
+    (Not having a declaration equation is similar to the expression being the default value for the component's type, which is constant.)
```

Comment:
Could you explain this sentence? Is "the default value for the component's type" the same as the start value?

Reply:
OK, I might be mistaken here, so let's consider an example:
```
model M
  parameter Boolean p; // Fine: parameter (regardless of fixed) doesn't have to be modified.
  constant Boolean c; // Fine: constant doesn't have to be modified.
end M;
```
The comments express my current understanding of the role of declaration equations thought of as modifications.  The value of the unmodified Booleans is false, coming from the `start` of the pseudocode definition of `Boolean` in 4.8.3 _Boolean Type_.

(System Modeler gives a warning for `p`, but inconsistently silently accepts `c`.)

Reply:
In a different context, @qlambert-pro just pointed me to the section that rules out the constant without declaration equation.  Quite surprisingly, it's stated in 3.8.1 _Constant Expressions_, rather than when describing constant component declaration in 4.4.4 _Component Variability Prefixes discrete, parameter, constant_:
> Components declared as `constant` shall have an associated declaration equation with a constant expression, if the constant is directly in the simulation model, or used in the simulation model.

Reply:
Here is another example:
```
type ThermodynamicTemperature = Real(start = 288.15);

model M
  parameter ThermodynamicTemperature T(start = 300);
...
end M;
```
The proposed text:
>(Not having a declaration equation is similar to the expression being the default value for the component's type, which is constant.)

suggests to me that the value of T should be 0. I am not sure if 288.15 can be considered  "the default value for the component's type" because it is modified. However, it would make more sense to take T.start (300) as the value of T, since this is the latest modification.

I am still not sure what "which is constant" refers to in the quoted sentence.

Reply:
I sincerely appreciate your effort to understand the proposal.  It really helps.

You are totally right regarding the role of `start` here for a `fixed = true` parameter without declaration equation.  It somehow slipped my mind that what we are talking about here is the special thing used, for example, in `Modelica.Electrical.Analog.Basic.Resistor` for the parameter `R`.  This isn't defined in either 3.8 (causing lack of symmetry with the case of `constant` without declaration equation) or 4.4.4 (which is the place I would try first if I didn't know where to find it), but in 8.6, item 3:
> Implicitly by using the `start` attribute for variables with `fixed = true`…
> - For a variable declared as `constant` or `parameter`, no equation is added to the initialization equations.

(The text looks outdated, since a constant is required to have a declaration equation.)

What it means here (see pushed change) is that `start` simply takes the role of the declaration equation in case it is missing.

It means that one can construct a normal parameter like this:
```
model M
  parameter Real T0 = 288.15 annotation(Evaluate = false); // Normal parameter, due to Evaluate = false.
  parameter Real T(start = T0); // Normal parameter, since no declaration equation and start isn't structural.
end M;
```

Reply:
I suppose this is considered first clarified, and then obsolete in view of the new fallback values.  Resolving.